### PR TITLE
Consider "main()" with typedef'd int return type as MainFunction

### PR DESCRIPTION
**Pull request by rak3-sh**
## Description

Currently, CodeQL considers `main` with an `int` return type as [MainFunction](https://github.com/github/codeql-coding-standards/blob/main/cpp/common/src/codingstandards/cpp/EncapsulatingFunctions.qll#L14). However, `main` can also be defined to return other typdef'd variants of int like `int32_t`. The lack of this consideration causes CodeQL to report false positives whenever a query wants to use the `MainFunction` class. An example of this is the issue [#646](https://github.com/github/codeql-coding-standards/issues/646) for [M0-1-10](https://github.com/github/codeql-coding-standards/blob/main/cpp/autosar/src/rules/M0-1-10/UnusedFunction.ql). This PR modifies the definition of `MainFunction` to also consider typedef'd variants of `int`.

## Change request type

 - [ ] Release or process automation (GitHub workflows, internal scripts)
 - [ ] Internal documentation
 - [ ] External documentation
 - [x] Query files (`.ql`, `.qll`, `.qls` or unit tests)
 - [ ] External scripts (analysis report or other code shipped as part of a release)

## Rules with added or modified queries

 - [ ] No rules added
 - [ ] Queries have been added for the following rules:
     - _rule number here_
 - [x] Queries have been modified for the following rules:
     - M0-1-10

## Release change checklist

A change note ([development_handbook.md#change-notes](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#change-notes)) is required for any pull request which modifies:

 - The structure or layout of the release artifacts.
 - The evaluation performance (memory, execution time) of an existing query.
 - The results of an existing query in any circumstance.

If you are only adding new rule queries, a change note is not required.

_**Author:**_ Is a change note required? 
  - [x] Yes
  - [ ] No

🚨🚨🚨
_**Reviewer:**_ Confirm that format of *shared* queries (not the .qll file, the
.ql file that imports it) is valid by running them within VS Code. 
  - [ ] Confirmed

_**Reviewer:**_ Confirm that either a change note is not required or the change note is required and has been added.
  - [ ] Confirmed

## Query development review checklist

For PRs that add new queries or modify existing queries, the following checklist should be completed by both the author and reviewer:

### Author

 - [x] Have all the relevant rule package description files been checked in?
 - [x] Have you verified that the metadata properties of each new query is set appropriately?
 - [x] Do all the unit tests contain both "COMPLIANT" and "NON_COMPLIANT" cases?
 - [x] Are the alert messages properly formatted and consistent with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide)?
 - [x] Have you run the queries on OpenPilot and verified that the performance and results are acceptable?<br />_As a rule of thumb, predicates specific to the query should take no more than 1 minute, and for simple queries be under 10 seconds. If this is not the case, this should be highlighted and agreed in the code review process._
 - [x] Does the query have an appropriate level of in-query comments/documentation?
 - [x] Have you considered/identified possible edge cases?
 - [x] Does the query not reinvent features in the standard library?
 - [x] Can the query be simplified further (not golfed!)

### Reviewer 

 - [x] Have all the relevant rule package description files been checked in?
 - [x] Have you verified that the metadata properties of each new query is set appropriately?
 - [ ] Do all the unit tests contain both "COMPLIANT" and "NON_COMPLIANT" cases?
 - [x] Are the alert messages properly formatted and consistent with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide)?
 - [x] Have you run the queries on OpenPilot and verified that the performance and results are acceptable?<br />_As a rule of thumb, predicates specific to the query should take no more than 1 minute, and for simple queries be under 10 seconds. If this is not the case, this should be highlighted and agreed in the code review process._
 - [x] Does the query have an appropriate level of in-query comments/documentation?
 - [x] Have you considered/identified possible edge cases?
 - [x] Does the query not reinvent features in the standard library?
 - [x] Can the query be simplified further (not golfed!)

### CODE REVIEW

File: cpp/autosar/test/rules/M0-1-10.1/test.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#include <cstdint>
```
Comment (lcartey) [inline cpp/autosar/test/rules/M0-1-10.1/test.cpp:1]: Could you please add a non-compliant function into this test case? e.g. a function which is never called.

**Review by lcartey — CHANGES_REQUESTED** [review]
One small change - other than that, looks good!

File: cpp/autosar/test/rules/M0-1-10.1/test.cpp (hunk 1 above)
Comment (rak3-sh) [reply cpp/autosar/test/rules/M0-1-10.1/test.cpp:1]: Done in [0c20066](https://github.com/github/codeql-coding-standards/pull/647/commits/0c200662d4ccb8de83792676e9d2f3d0030d7cd2)

**Review by lcartey — APPROVED** [review]
